controllers/voteController.js

```diff
@@ -1,20 +1,32 @@
 var { db } = require("../models/index.js");
-var { VoteService } = require("../services/index.js");
+var { VoteService, ProjectService } = require("../services/index.js");
 var voteService = new VoteService(db);
+var projectService = new ProjectService(db);
 var { successResponse } = require("../utilities/response.js");
+var createError = require("../utilities/createError");
 
 async function toggleUpvote(req, res) {
   const { projectId } = req.params;
   const { id: userId } = req.user;
 
+  const project = await projectService.getOneId(projectId);
+  if (!project) {
+    throw createError({
+      message: "Project not found",
+      status: "not found",
+      statusCode: 404,
+      errors: { projectId },
+    });
+  }
```

Comment:
The update to the projectService error handling means this will throw a 404 if nothing is found.
```js
  async getOneId(id, currentUserId = null) {
    const queryBuilder = new ProjectQueryBuilder().withVotes().withTypes().withUser().filterById(id);

    const project = await this.client.query(queryBuilder.buildListQuery(), {
      replacements: {
        projectId: id,
        currentUserId,
        limit: 1,
        offset: 0,
      },
      type: this.client.QueryTypes.SELECT,
    });

    // Since you’re expecting one, not many:
    if (project.length === 0) {
      throw createError({
        message: "Project not found",
        statusCode: 404,
      });
    }
    return project[0];
  }
  ```
  So this check can be removed
  ```suggestion
  const project = await projectService.getOneId(projectId);
```